### PR TITLE
images: use k8s-staging-test-infra/gcb-docker-gcloud

### PR DESCRIPTION
**Pull request by spiffxp**
Related:
- Part of: https://github.com/kubernetes/k8s.io/issues/1523

### CODE REVIEW

File: docker/cloudbuild.yaml [hunk 1]
```diff
@@ -13,7 +13,7 @@ options:
   # job builds a multi-arch docker image for amd64,arm64 and windows 1809, 1903, 1909, 2004.
   machineType: 'N1_HIGHCPU_8'
 steps:
-  - name: 'gcr.io/k8s-testimages/gcb-docker-gcloud:v20210722-085d930'
+  - name: 'gcr.io/k8s-staging-test-infra/gcb-docker-gcloud:v20211008-60e346af'
```
Comment (aramase) [inline docker/cloudbuild.yaml:16]: Looks like this image tag doesn't exist?

```bash
➜ docker buildx imagetools inspect gcr.io/k8s-staging-test-infra/gcb-docker-gcloud:v20211008-60e346af
INFO[0000] trying next host - response was http.StatusNotFound  host=gcr.io
error: gcr.io/k8s-staging-test-infra/gcb-docker-gcloud:v20211008-60e346af: not found

➜ docker pull gcr.io/k8s-staging-test-infra/gcb-docker-gcloud:v20211008-60e346af
Error response from daemon: manifest for gcr.io/k8s-staging-test-infra/gcb-docker-gcloud:v20211008-60e346af not found: manifest unknown: Failed to fetch "v20211008-60e346af" from request "/v2/k8s-staging-test-infra/gcb-docker-gcloud/manifests/v20211008-60e346af".
```

**Review by aramase — COMMENTED** [review]
/hold

File: docker/cloudbuild.yaml (hunk 1 above)
Comment (spiffxp) [reply docker/cloudbuild.yaml]: Good catch, tag has been updated

**Review by aramase — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Pull request merged**